catalog/app/utils/spreadsheets/spreadsheets.ts

```diff
@@ -27,7 +34,16 @@ export function parseSpreadsheet(
   const rows = xlsx.utils.sheet_to_json<MetadataValue>(sheet, {
     header: 1,
   })
-  return rowsToJson(transpose ? R.transpose(rows) : rows)
+  const maxSize = rows.reduce((memo, row) => R.max(memo, row.length), 0)
+  return pipeThru(rows)(
+    R.without([[]]),
+    R.map((row: any[]) => {
+      const nullsTail = R.repeat(null, maxSize - row.length)
+      return R.concat(row, nullsTail)
+    }),
+    R.when(() => transpose, R.transpose),
```

Comment:
since the computation doesnt depend on the actual values, smth like this seems more logical: `transpose ? R.transpose : R.identity,`
tho it doesnt make much difference